src/components/fs/FS.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <cstdint>
+#include "drivers/SpiNorFlash.h"
+#include <littlefs/lfs.h>
+
+namespace Pinetime {
+  namespace Controllers {
+    class FS {
+        public:
+            FS(Pinetime::Drivers::SpiNorFlash&);
+
+            Pinetime::Drivers::SpiNorFlash& mDriver;
```

Comment:
My idea was to simplify, putting it as private forced the creation of functions to call SpiNorFlash, on the other hand we have the possibility of calling SpiNorFlash where we are already using FS.

Reply:
I agree that putting this field as private looks a bit cumbersome and adds an indirection that can be easily avoided.

However, my concern is about the encapsulation and the API exposed by this class FS.

When a field or a method is public, users of this class (`Controllers::Settings`, probably many others in the near future) should be able to use them without breaking the system. What if, in the future, someone writes some code that directly writes to the spi flash and corrupt the filesystem? Of course, we'll probably notice that during the review process, but it would be better if the API of the class did not allow that from the beginning.

Also in this case, `Settings` only needs to know it's saving the settings into a file system. It doesn't need to know this file system lives in an `SpiNorFlash`, in the internal flash or in RAM. If the underlying memory is exposed, `Settings` (or another Controller/App) might use this field for any reason, and would not be able to change the implementation of `FS` and `SpiNorFlash` that easily, because they are used at a lot more places than needed.

I know it's a bit theoretical, and there is very little point anyone would want to write these things right now, but you never know how the code will evolve. I always try to write my classes so that they expose a clear API and as few fields as possible, this always helps maintaining the code in the long term. It'll also help us the day we want to port the code to another platform with another memory configuration, for example.

Reply:
I got the idea and the concern, I'm going to do the functions to make the object private.

Reply:
Thanks :+1:

Reply:
Changes made in FS to make SpiNorFlash private.